### PR TITLE
tiup: add TiUP DM introduction and docs for command `dm-restart` and `dm-patch`

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add 3 docs:

- tiup/tiup-component-dm.md
- tiup/tiup-component-dm-restart.md
- tiup/tiup-component-dm-patch.md

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/4868
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by CharLotteiu** [comment]
Waiting for other `tiup-component-dm-xx.md` files to be merged to fix the dead links. This PR should be the last one to be merged.

File: tiup/tiup-component-dm-patch.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    - Run `find .` to view the file structure in the temporary package directory.
+    - Copy the binary files or configuration files to the corresponding locations in the temporary directory.
+    - Run `tar czf /tmp/${component}-hotfix-${os}-${arch}.tar.gz *` to pack the files in the temporary directory.
+    - Finally, you can use `/tmp/${component}-hotfix-${os}-${arch}.tar.gz` as the `<package-path>` in the `tiup cluster patch` command.
+
+## Options
+
+### --overwrite
+
+- After you patch a certain component (such as TiDB or TiKV), when the tiup cluster scales out the component, tiup-dm uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specified the option `--overwrite` in the command.
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:36]:
```suggestion
- After you patch a certain component (such as TiDB or TiKV), when the tiup cluster scales out the component, tiup-dm uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specify the option `--overwrite` in the command.
```

File: tiup/tiup-component-dm-patch.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- Data type: `BOOLEAN`
+- This option is disabled by default with the `false` value. To enable this option, add this option to the command, and either pass the `true` value or do not pass any value.
+
+### -N, --node
+
+- Specifies nodes to be replaced. The value of this option is a comma-separated list of node IDs. You can get the node ID from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:42]:
```suggestion
- Specifies the nodes to be replaced. The value of this option is a comma-separated list of node IDs. You can get the node IDs from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```

File: tiup/tiup-component-dm-patch.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+- Data type: `STRINGS`
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:43]:
```suggestion
- Data type: `STRING`
```

File: tiup/tiup-component-dm-patch.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+- If this option is not specified, TiUP does not select any nodes to replace by default.
+
+> **Note:**
+>
+> If the option `-R, --role` is specified at the same time, TiUP then replaces service nodes that match both the requirements of `-N, --node` and `-R, --role`.
+
+### -R, --role
+
+- Specified roles to be replaced. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:52]:
```suggestion
- Specified the roles to be replaced. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```

File: tiup/tiup-component-dm-patch.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+- Data type: `STRINGS`
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:53]:
```suggestion
- Data type: `STRING`
```

File: tiup/tiup-component-dm-patch.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+- If this option is not specified, TiUP does not select any roles to replace by default.
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:54]: The Chinese version of this sentence has changed to "默认为 []，表示所有节点". Please refer to https://github.com/pingcap/docs-cn/pull/5765.

File: tiup/tiup-component-dm-patch.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+- After you patch a certain component (such as TiDB or TiKV), when the tiup cluster scales out the component, tiup-dm uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specified the option `--overwrite` in the command.
+- Data type: `BOOLEAN`
+- This option is disabled by default with the `false` value. To enable this option, add this option to the command, and either pass the `true` value or do not pass any value.
+
+### -N, --node
+
+- Specifies nodes to be replaced. The value of this option is a comma-separated list of node IDs. You can get the node ID from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
+- Data type: `STRINGS`
+- If this option is not specified, TiUP does not select any nodes to replace by default.
```
Comment (qiancai) [inline tiup/tiup-component-dm-patch.md:44]: The Chinese version of this sentence has changed to "默认为 []，表示所有节点". Please refer to https://github.com/pingcap/docs-cn/pull/5765.

File: tiup/tiup-component-dm-restart.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+tiup dm restart <cluster-name> [flags]
+```
+
+`<cluster-name>`: the name of the cluster to operate on. If you forget the cluster name, you can check it with the [cluster list](/tiup/tiup-component-cluster-list.md) command.
+
+## Options
+
+### -N, --node
+
+- Specifies the nodes to be restarted. The value of this option is a comma-separated list of node IDs. You can get the node ID from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```
Comment (qiancai) [inline tiup/tiup-component-dm-restart.md:25]:
```suggestion
- Specifies the nodes to be restarted. The value of this option is a comma-separated list of node IDs. You can get the node IDs from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```

File: tiup/tiup-component-dm-restart.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+- Data type: `STRINGS`
+- If this option is not specified, TiUP restarts all nodes by default.
+
+> **Note:**
+>
+> If the option `-R, --role` is specified at the same time, TiUP restarts service nodes that match both the requirements of `-N, --node` and `-R, --role`.
+
+### -R, --role
+
+- Specified roles of nodes to be restarted. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```
Comment (qiancai) [inline tiup/tiup-component-dm-restart.md:35]:
```suggestion
- Specified the roles of nodes to be restarted. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-cluster-display.md).
```

File: tiup/tiup-component-dm-patch.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+## Syntax
+
+```shell
+tiup dm patch <cluster-name> <package-path> [flags]
+```
+
+- `<cluster-name>`: The name of the cluster to be operated.
+- `<package-path>`: The path to the binary package used for replacement. The steps to generate the package are as follows:
+    - Determine the name `${component}` (dm-master, dm-worker ...) of the component to be replaced (tidb, tikv, pd...), the `${version}` of the component (v2.0.0, v2.0.1 ...), and the operating system `${os}` and platform `${arch}` on which the component runs.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply tiup/tiup-component-dm-patch.md:23]:
```suggestion
    - Determine the name `${component}` (dm-master, dm-worker ...) of the component to be replaced, the `${version}` of the component (v2.0.0, v2.0.1 ...), and the operating system `${os}` and platform `${arch}` on which the component runs.
```

File: tiup/tiup-component-dm-patch.md (hunk 7 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-patch.md:44]:
```suggestion
- If this option is not specified, TiUP selects all nodes to replace by default.
```

File: tiup/tiup-component-dm-patch.md (hunk 2 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-patch.md:42]:
```suggestion
- Specifies nodes to be replaced. The value of this option is a comma-separated list of node IDs. You can get the node IDs from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md) returned by the `tiup cluster display` command..
```

File: tiup/tiup-component-dm-patch.md (hunk 4 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-patch.md:52]:
```suggestion
- Specified the roles to be replaced. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-cluster-display.md) returned by the `tiup cluster display` command.
```

File: tiup/tiup-component-dm-patch.md (hunk 6 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-patch.md:54]:
```suggestion
- If this option is not specified, TiUP selects all roles to replace by default.
```

File: tiup/tiup-component-dm-restart.md (hunk 8 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-restart.md:25]:
```suggestion
- Specifies the nodes to be restarted. The value of this option is a comma-separated list of node IDs. You can get the node IDs from the first column of the [cluster status table](/tiup/tiup-component-cluster-display.md) returned by the `tiup cluster display` command.
```

File: tiup/tiup-component-dm-restart.md (hunk 9 above)
Comment (CharLotteiu) [reply tiup/tiup-component-dm-restart.md:35]:
```suggestion
- Specified the roles of nodes to be restarted. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the [cluster status table](/tiup/tiup-component-dm-display.md) returned by the `tiup dm display` command.
```

File: tiup/tiup-component-dm.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+### -y, --yes
+
+- Skips the secondary confirmation of all risky operations. It is not recommended to use this option unless you use a script to call TiUP.
+- Data type:`BOOLEAN`
+- This option is disabled by default with the `false` value. To enable this option, add this option to the command, and either pass the `true` value or do not pass any value.
+
+### -v, --version
+
+- Prints the current version of TiUP Cluster.
```
[the remark this answers is not in the document]
Comment (CharLotteiu) [reply tiup/tiup-component-dm.md:51]:
```suggestion
- Prints the current version of TiUP DM.
```

File: tiup/tiup-component-dm-patch.md [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+    - Determine the name `${component}` (dm-master, dm-worker ...) of the component to be replaced, the `${version}` of the component (v2.0.0, v2.0.1 ...), and the operating system `${os}` and platform `${arch}` on which the component runs.
+    - Download the current component package using the command `wget https://tiup-mirrors.pingcap.com/${component}-${version}-${os}-${arch}.tar.gz -O /tmp/${component}-${version}-${os}-${arch}.tar.gz`.
+    - Run `mkdir -p /tmp/package && cd /tmp/package` to create a temporary directory to pack files.
+    - Run `tar xf /tmp/${component}-${version}-${os}-${arch}.tar.gz` to unpack the original binary package.
+    - Run `find .` to view the file structure in the temporary package directory.
+    - Copy the binary files or configuration files to the corresponding locations in the temporary directory.
+    - Run `tar czf /tmp/${component}-hotfix-${os}-${arch}.tar.gz *` to pack the files in the temporary directory.
+    - Finally, you can use `/tmp/${component}-hotfix-${os}-${arch}.tar.gz` as the `<package-path>` in the `tiup cluster patch` command.
```
Comment (CharLotteiu) [inline tiup/tiup-component-dm-patch.md:30]:
```suggestion
    - Finally, you can use `/tmp/${component}-hotfix-${os}-${arch}.tar.gz` as the `<package-path>` in the `tiup dm patch` command.
```

**Review by CharLotteiu — COMMENTED** [review]
Change `tiup cluster` to `tiup dm`

File: tiup/tiup-component-dm-patch.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+- Data type: `STRING`
+- If this option is not specified, TiUP selects all nodes to replace by default.
+
+> **Note:**
+>
+> If the option `-R, --role` is specified at the same time, TiUP then replaces service nodes that match both the requirements of `-N, --node` and `-R, --role`.
+
+### -R, --role
+
+- Specified the roles to be replaced. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the cluster status table returned by the `[tiup dm display](/tiup/tiup-component-dm-display.md)` command.
```
Comment (CharLotteiu) [inline tiup/tiup-component-dm-patch.md:57]:
```suggestion
- Specifies the roles to be replaced. The value of this option is a comma-separated list of the roles of the nodes. You can get the roles of the nodes from the second column of the cluster status table returned by the `[tiup dm display](/tiup/tiup-component-dm-display.md)` command.
```

File: tiup/tiup-component-dm-patch.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+- Run `find .` to view the file structure in the temporary package directory.
+- Copy the binary files or configuration files to the corresponding locations in the temporary directory.
+- Run `tar czf /tmp/${component}-hotfix-${os}-${arch}.tar.gz *` to pack the files in the temporary directory.
+- Finally, you can use `/tmp/${component}-hotfix-${os}-${arch}.tar.gz` as the value of `<package-path>` in the `tiup dm patch` command.
+
+## Options
+
+### --overwrite
+
+- After you patch a certain component (such as TiDB or TiKV), when the tiup cluster scales out the component, tiup-dm uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specify the option `--overwrite` in the command.
```
Comment (AstroProfundis) [inline tiup/tiup-component-dm-patch.md:41]:
```suggestion
- After you patch a certain component (such as dm-worker), when the tiup-dm scales out the component, tiup-dm uses the original component version by default. To use the version that you patch when the cluster scales out in the future, you need to specify the option `--overwrite` in the command.
```

File: tiup/tiup-component-dm.md [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+
+### --ssh
+
+- Specifies the SSH client to connect to the remote end (the machine where the TiDB service is deployed) for the command execution.
+- Data type:`STRING`
+- Support values:
+
+    - `builtin`: Uses the built-in easyssh client of tiup-cluster as the SSH client.
+    - `system`: Uses the default SSH client of the current operating system.
+    - `none`: The SSH client is not used. The deployment is only for the current machine.
```
Comment (lucklove) [inline tiup/tiup-component-dm.md:27]:
```suggestion
    - `none`: No SSH client is used. The deployment is only for the current machine.
```

**Comment by lucklove** [comment]
Rest LGTM

**Pull request merged**